service/controller/v1/resource/chartconfig/current.go

```diff
@@ -29,25 +30,29 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	//	customResource, err := key.ToCustomResource(obj)
-	//	if err != nil {
-	//		return nil, microerror.Mask(err)
-	//	}
+	customResource, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
 
-	var chartConfigCR *v1alpha1.ChartConfig
+	var chartConfigCRs []*v1alpha1.ChartConfig
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "finding current state")
 
-		m, err := r.g8sClient.CoreV1alpha1().ChartConfigs(r.namespace).Get("", metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find current state")
-		} else if err != nil {
+		o := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", key.LabelReleaseVersion, key.ReleaseVersion(customResource)),
+		}
+		list, err := r.g8sClient.CoreV1alpha1().ChartConfigs(r.namespace).List(o)
+		if err != nil {
 			return nil, microerror.Mask(err)
-		} else {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "found current state")
-			chartConfigCR = m
 		}
+
+		for _, c := range list.Items {
+			chartConfigCRs = append(chartConfigCRs, &c)
+		}
```

Comment:
I would like to deal with consistent types. The desired state computes pointers. The list items are values. Don't know.